patches/systemd-utils/0001-Add-managarm-support.patch

```diff
@@ -0,0 +1,906 @@
+-#ifndef __UAPI_DEF_IF_NET_DEVICE_FLAGS_LOWER_UP_DORMANT_ECHO
+-#define __UAPI_DEF_IF_NET_DEVICE_FLAGS_LOWER_UP_DORMANT_ECHO 1
+-#endif
+-
+-/* Definitions for in.h */
+-#ifndef __UAPI_DEF_IN_ADDR
+-#define __UAPI_DEF_IN_ADDR		1
+-#endif
+-#ifndef __UAPI_DEF_IN_IPPROTO
+-#define __UAPI_DEF_IN_IPPROTO		1
+-#endif
+-#ifndef __UAPI_DEF_IN_PKTINFO
+-#define __UAPI_DEF_IN_PKTINFO		1
+-#endif
+-#ifndef __UAPI_DEF_IP_MREQ
+-#define __UAPI_DEF_IP_MREQ		1
+-#endif
+-#ifndef __UAPI_DEF_SOCKADDR_IN
+-#define __UAPI_DEF_SOCKADDR_IN		1
+-#endif
+-#ifndef __UAPI_DEF_IN_CLASS
+-#define __UAPI_DEF_IN_CLASS		1
+-#endif
+-
+-/* Definitions for in6.h */
+-#ifndef __UAPI_DEF_IN6_ADDR
+-#define __UAPI_DEF_IN6_ADDR		1
+-#endif
+-#ifndef __UAPI_DEF_IN6_ADDR_ALT
+-#define __UAPI_DEF_IN6_ADDR_ALT		1
+-#endif
+-#ifndef __UAPI_DEF_SOCKADDR_IN6
+-#define __UAPI_DEF_SOCKADDR_IN6		1
+-#endif
+-#ifndef __UAPI_DEF_IPV6_MREQ
+-#define __UAPI_DEF_IPV6_MREQ		1
+-#endif
+-#ifndef __UAPI_DEF_IPPROTO_V6
+-#define __UAPI_DEF_IPPROTO_V6		1
+-#endif
+-#ifndef __UAPI_DEF_IPV6_OPTIONS
+-#define __UAPI_DEF_IPV6_OPTIONS		1
+-#endif
+-#ifndef __UAPI_DEF_IN6_PKTINFO
+-#define __UAPI_DEF_IN6_PKTINFO		1
+-#endif
+-#ifndef __UAPI_DEF_IP6_MTUINFO
+-#define __UAPI_DEF_IP6_MTUINFO		1
+-#endif
+-
+-/* Definitions for ipx.h */
+-#ifndef __UAPI_DEF_SOCKADDR_IPX
+-#define __UAPI_DEF_SOCKADDR_IPX			1
+-#endif
+-#ifndef __UAPI_DEF_IPX_ROUTE_DEFINITION
+-#define __UAPI_DEF_IPX_ROUTE_DEFINITION		1
+-#endif
+-#ifndef __UAPI_DEF_IPX_INTERFACE_DEFINITION
+-#define __UAPI_DEF_IPX_INTERFACE_DEFINITION	1
+-#endif
+-#ifndef __UAPI_DEF_IPX_CONFIG_DATA
+-#define __UAPI_DEF_IPX_CONFIG_DATA		1
+-#endif
+-#ifndef __UAPI_DEF_IPX_ROUTE_DEF
+-#define __UAPI_DEF_IPX_ROUTE_DEF		1
+-#endif
+-
+-/* Definitions for xattr.h */
+-#ifndef __UAPI_DEF_XATTR
+-#define __UAPI_DEF_XATTR		1
+-#endif
+-
+-#endif /* __GLIBC__ */
+-
+-#endif /* _UAPI_LIBC_COMPAT_H */
++#endif /* _LINUX_LIBC_COMPAT_H */
+diff --git a/src/basic/missing_syscall.h b/src/basic/missing_syscall.h
+index e2cd8b4..5c0bc20 100644
+--- a/src/basic/missing_syscall.h
++++ b/src/basic/missing_syscall.h
+@@ -11,7 +11,7 @@
+ #include <linux/time_types.h>
+ #endif
+ #include <signal.h>
+-#include <sys/syscall.h>
++// #include <sys/syscall.h>
+ #include <sys/types.h>
+ #include <sys/wait.h>
+ #include <unistd.h>
+@@ -35,7 +35,7 @@
+ 
+ #if !HAVE_FCHMODAT2
+ static inline int missing_fchmodat2(int dirfd, const char *path, mode_t mode, int flags) {
+-#  ifdef __NR_fchmodat2
++#  if defined(__NR_fchmodat2) && !defined(__managarm__)
+         return syscall(__NR_fchmodat2, dirfd, path, mode, flags);
+ #  else
+         errno = ENOSYS;
+@@ -50,7 +50,11 @@ static inline int missing_fchmodat2(int dirfd, const char *path, mode_t mode, in
+ 
+ #if !HAVE_PIVOT_ROOT
+ static inline int missing_pivot_root(const char *new_root, const char *put_old) {
++#ifdef __managarm__
++        assert(!"unimplemented pivot_root");
++#else
```

Comment:
`assert()` can be compiled out. Ideally, we log to `/dev/helout` instead (IIRC systemd redirects its stdout/stderr). We can add a helper function to this file for that purpose.

Reply:
Yep, systemd does redirect that. I agree that we should log here.